### PR TITLE
Feature/stale channel index bucket test

### LINKED ISSUE #813: Functional test to detect stale channel index bucket
Steps to reproduce:
- Bring up single Sync Gateway node, backed by Couchbase Server.  
- Configure such that the primary bucket and the channel index bucket are different (which is the norm)
- Add several documents
- Shutdown Sync Gateway
- Flush the primary bucket, but **do not touch** the channel index bucket
- Start Sync Gateway
- Make sure that Sync Gateway starts without errors.

See https://github.com/couchbase/sync_gateway/issues/1133

### PR DESCRIPTION
**Pull request by sethrosetter**
- Fixes #813 
- Rename keywords.CouchbaseServer -> keywords.couchbaseserver
- Add delete_bucket method
- Add get_ram_per_bucket method + various RAM calculation clean up
- Add bucket deletion verification

### CODE REVIEW

**Comment by sethrosetter** [comment]
test failing due to - https://github.com/couchbase/sync_gateway/issues/1133

**Comment by sethrosetter** [comment]
Review couchbaseserver.py changes carefully.

**Comment by tleyden** [comment]
@sethrosetter can you rebase this and resolve the conflicts?

File: keywords/couchbaseserver.py [hunk 1]
```diff
@@ -161,28 +174,49 @@ def get_available_ram(self):
...
+        log_info("total_avail_ram_mb: {}".format(total_avail_ram_mb))
+        return total_avail_ram_mb
+
+    def _get_effective_ram_mb(self):
+        """ Return the amount of effective RAM ((total RAM * muliplier) - n1ql ram allocation)
+        Given a total amount of ram
+        """
+
+        ram_multiplier = 0.80
+        n1ql_indexer_ram_mb = 512
```
Comment (raghusarangapani) [inline keywords/couchbaseserver.py]: Can you add some comments to indicate how you came up with these numbers?

**Comment by sethrosetter** [comment]
Waiting to rebase / merge until this is https://github.com/couchbase/sync_gateway/issues/1133 fixed.

**Comment by tleyden** [comment]
I just pushed a few more commits to this, and the test now passes when run against https://github.com/couchbaselabs/sync-gateway-accel/pull/59

Test output here: https://github.com/couchbase/sync_gateway/issues/1133#issuecomment-275293120

**Comment by tleyden** [comment]
@sethrosetter I think we left this at the point where you said you would rebase it and merge it

File: keywords/couchbaseserver.py (hunk 1 above)
Comment (sethrosetter) [inline keywords/couchbaseserver.py]: @tleyden Where did you get those numbers? I just ported it.

**Comment by sethrosetter** [comment]
@tleyden Rebased and added addition verification on bucket deletions. Can you re-review couchbaseserver.py?

File: keywords/couchbaseserver.py (hunk 1 above)
Comment (tleyden) [inline keywords/couchbaseserver.py]: I added comments

Comment (tleyden) [inline keywords/couchbaseserver.py]: @adamcfraser do you remember the details around `n1ql_indexer_ram_mb = 512`

File: keywords/couchbaseserver.py [hunk 2]
```diff
@@ -200,8 +200,12 @@ def _get_effective_ram_mb(self):
         Given a total amount of ram
         """
 
+        # Leave 20% of RAM available for the underlying OS
         ram_multiplier = 0.80
+
+        # Needed for ability to add a N1QL indexing bucket.  @adamcfraser do you remember why we need this?
```
Comment (sethrosetter) [inline keywords/couchbaseserver.py]: I pushed a commit to master with a comment for this

Comment (tleyden) [inline keywords/couchbaseserver.py]: Awesome!  Thanks @raghusarangapani for raising that

**Pull request merged**